langchain/src/vectorstores/qdrant.ts

```diff
@@ -32,17 +33,24 @@ export class QdrantVectorStore extends VectorStore {
     super(embeddings, args);
 
     const url = args.url ?? getEnvironmentVariable("QDRANT_URL");
+    const apiKey = args.apiKey ?? getEnvironmentVariable("QDRANT_API_KEY")
 
     if (!args.client && !url) {
       throw new Error("Qdrant client or url address must be set.");
     }
 
-    this.client =
-      args.client ||
-      new QdrantClient({
-        url,
-      });
-
+    // apiKey is necessary for Qdrant Cloud, but not self hosted
+    if (!args.client && apiKey) {
+        this.client =
+        args.client ||
+        new QdrantClient({
+            url,
+            apiKey,
+        });
```

Comment:
Empty api key won't hurt, you can drop the condition and just use the version with api key param

Reply:
Done!